src/main/java/controllers/UserProfileController.java

```diff
@@ -48,36 +47,38 @@ else if (action == ShellAction.CUSTOMIZEPROFILE) {
      * Return the choices available for user
      */
     private String getCustomizeProfileScreen() {
-        return this.customizeProfileScreen;
+        return "Select which option you'd like to customize:\n" +
+                    "0 Username\n" +
+                    "1 Password\n" +
+                    "2 Bio\n" +
+                    "3 Posts\n" +
+                    "4 Following list\n";
     }
 
     /**
-     * getUserInformation, Displays the user from the username given, return to main page if user not found
+     * Run the browse profile logic
      */
     public void runBrowseProfile() {
-        // Browse user's profile, given String username:
-        // run a helper method (checkUserToBrowse) to call Database manager to check whether the username belongs
-        // to a user in the database
-        // run findUser to get the user that owns the username from the database
         try {
             // display all the usernames in the database
-            // runDisplayAllUsers()
+            this.runDisplayAllUsers();
             String userToBrowse = this.inOut.getInput("Enter the username of the person you'd like to view: ");
-            if (this.userManager.checkUserToBrowse(userToBrowse) ||
-                    userToBrowse.equals(this.loginManager.getCurrUser().getUsername())) {
+            if (this.userManager.checkUserToBrowse(userToBrowse) &&
+                    !(userToBrowse.equals(this.loginManager.getCurrUser().getUsername()))) {
                 User targetUser = this.userManager.findUser(userToBrowse);
                 this.inOut.setOutput(this.userManager.runBrowseOtherProfile(targetUser));
-                // Give the choice of look following user or looking into user's post
-                int choice = Integer.parseInt(this.inOut.getInput(this.otherUserScreen));
+                // Give the choice of following user or looking into user's post
+                String otherUserScreen = "Select which action you'd like to take:\n" +
+                        "0 Follow user\n" +
+                        "1 Browse user's posts\n";
```

Comment:
what if user inputs an invalid option? Should we account for that?

Reply:
an error msg will pop up and it will return the user to the main menu